### ISSUE TITLE
A11y: Insufficient accessibility of the search form field for screen readers

### ISSUE THREAD

**Issue by blacksheep--** [shopware/platform#7093]
*PHP Version:* 8.2   | *Shopware Version:* 6.6.9.0  | *Affected area / extension:* Platform(Default) 

 ----

*_Actual behaviour:_*

The search form field is not programmatically designed as an “input field with suggestion function”. After entering a phrase, screen reader users are not informed that search results are available. And even if screen reader users actively query whether there are search suggestions (search results) using the down arrow, nothing is read out.

This construct is difficult for blind people to understand when using the site for the first time.

*_Expected behaviour:_*

As soon as search results are available, the existence of the results must be indicated using `aria-live`. The [datalist pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist) can be implemented to make the search hits accessible from the input field.

Further Resources:
- [github alphagov/accessible-autocomplete](https://github.com/alphagov/accessible-autocomplete)
- [Accessible Autocomplete examples](https://alphagov.github.io/accessible-autocomplete/examples/)

*_How to reproduce:_*

Enable `ACCESSIBILITY_TWEAKS` Feature-Flag

1. enter the search field and type until suggestions appear

**Comment by shopwareBot** [comment]
[public] Shopware automated response: This issue is linked to the internal issue: **NEXT-40848**



[created from NEXT-40848, comment 541828]

**Comment by shopwareBot** [comment]
[public] 

 > Priority changed to: **High**

 Thank you for your report. Based on the information in this report, we have decided to investigate this issue in more detail and are planning to provide a fix for this issue. Although we are always eager to solve such problems as soon as we can, we can't give you a more specific time frame right now. We appreciate your patience and thank you for your understanding.



[created from NEXT-40848, comment 541995]

**Comment by shopwareBot** [comment]
[public] 

 > Priority changed to: **Low**

 Thank you for your report. After careful consideration, we've concluded that this issue does not require immediate attention or isn't something we can address right now. We prioritize openness and clear communication with our community, so we want to be transparent. We may not resolve all reported issues, and this one might remain unresolved. To stay effective, we periodically clean our backlog, and this issue could be closed in the future.

 For now, we'll keep this ticket open to invite further feedback from the community in case we've underestimated its impact and to allow us to revisit it later.



[created from NEXT-40848, comment 541996]
